arraycontext/impl/pytato/__init__.py

```diff
@@ -271,9 +289,26 @@ def __init__(
             representation. This interface should be considered
             unstable.
         """
+        import pyopencl as cl
+
         if allocator is not None and use_memory_pool is not None:
             raise TypeError("may not specify both allocator and use_memory_pool")
 
+        self.profile_kernels = profile_kernels
+
+        if profile_kernels:
+            if not queue.properties & cl.command_queue_properties.PROFILING_ENABLE:
+                raise RuntimeError("Profiling was not enabled in the command queue. "
+                 "Please create the queue with "
+                 "cl.command_queue_properties.PROFILING_ENABLE.")
```

Comment:
Is the extra flag flexibility needed? i.e. could the context always profile if given a profiling enabled queue? (I'm fine with keeping it, just curious if it came from some use case).

Reply:
For `self.profile_kernels`, my idea was that the user can control even during execution whether collecting profiling data (and its associated overhead) is actually done.